libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -551,6 +576,20 @@ export class FunctionUtils {
         }, verify);
     }
 
+    /**
+     *
+     * @param args An array of arguments.
+     * @param locale A locale string
+     * @param maxArgsLength The max length of a given function.
+     */
+    public static determineLocale(args: any[], maxArgsLength: number, locale = 'en-us'): string {
+        if (args.length === maxArgsLength) {
+            locale = args[maxArgsLength - 1] as string;
```

Comment:
We may as well be extra careful and ensure `typeof args[maxArgsLength - 1] === 'string'` to remove the need for casting.

Reply:
Though `typeof args[maxArgsLength - 1] === 'string'` is added. The typescript still throws an error if we remove the cast.
So I kept the cast and add this type checking in the condition.